balancer/rls/internal/builder.go

```diff
@@ -21,16 +21,35 @@
 // Package rls implements the RLS LB policy.
 package rls
 
+import (
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/internal/grpcsync"
+)
+
 const rlsBalancerName = "rls"
 
+func init() {
+	balancer.Register(&rlsBB{})
+}
+
 // rlsBB helps build RLS load balancers and parse the service config to be
 // passed to the RLS load balancer.
-type rlsBB struct {
-	// TODO(easwars): Implement the Build() method and register the builder.
-}
+type rlsBB struct{}
 
 // Name returns the name of the RLS LB policy and helps implement the
 // balancer.Balancer interface.
 func (*rlsBB) Name() string {
 	return rlsBalancerName
 }
+
+func (*rlsBB) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
+	lb := &rlsBalancer{
+		done:       grpcsync.NewEvent(),
+		cc:         cc,
+		opts:       opts,
+		lbCfg:      &lbConfig{},
```

Comment:
It was initially `nil`, which meant that before we compare the old and new configs in `handleClientConnUpdate`, we had to do a `nil` check (because the first time we receive the service config, old config would be `nil`).

Reply:
If `lbCfg.Equal` works on `nil` receiver, the old can be nil.

Can `newCfg` be nil? If `newCfg` is never nil, `newCfg.Equal(oldCfg)`.

But this is a minor problem. Keeping it this way (never nil) is OK.

Reply:
Ack.